compiler/modules/types/list.bal

```diff
@@ -359,6 +362,17 @@ function fixedArrayShallowCopy(FixedLengthArray array) returns FixedLengthArray
     return { initial: shallowCopyTypes(array.initial), fixedLength: array.fixedLength };
 }
 
+function listConjunctionMaxInitialLength(Context cx, Conjunction? con) returns int {
```

Comment:
Would it be an improvement to change this function to create a record ListConjunction, which is like Conjunction but contains a ListAtomicType instead of an atom, and has a maxInitialLen field which is the maximum of the initial length of this and all conjunctions? (This would be easiest to write recursively.) Then you don't need the maxInitialLen argument to listInhabited and you don't need the do to Atom->ListAtomicType conversion twice. You would then compute maxInitialLen in listInhabited from members and neg.maxInitialLen.